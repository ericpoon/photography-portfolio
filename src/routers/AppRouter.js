import React from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';
import DashboardPage from '../components/DashboardPage';
import NotFoundPage from '../components/NotFoundPage';
import HomePage from '../components/HomePage';
import PortfolioPage from '../components/PortfolioPage';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';
import RouteWithHeader from './RouteWithHeader';

export const history = createHistory();

const AppRouter = () => (
  <Router history={history}>
    <div>
      <Switch>
        <Route path={'/'} component={HomePage} exact />
        <RouteWithHeader path={'/portfolio'} component={PortfolioPage} exact />
        {/* <PublicRoute path={'/'} component={HomePage} exact /> */}
        {/* <PrivateRoute path={'/dashboard'} component={DashboardPage} exact /> */}
        <Route component={NotFoundPage} />
      </Switch>
    </div>
  </Router>
);

export default AppRouter;
