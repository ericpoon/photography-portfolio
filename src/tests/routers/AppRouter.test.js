import React from 'react';
import { shallow } from 'enzyme';
import AppRouter from '../../routers/AppRouter';

let wrapper,
  routes;

beforeEach(() => {
  wrapper = shallow(<AppRouter />);
  routes = wrapper.find('Switch').children();
});

it('renders AppRouter', () => {
  expect(wrapper).toMatchSnapshot();
});

it('has a unrestricted route for 404 fallback page', () => {
  const route = routes.filterWhere(n => !n.prop('path'));
  expect(route).toHaveLength(1);
  expect(route.name()).toBe('Route');
});

it('has a unrestricted route for home page', () => {
  const route = routes.filterWhere(n => n.prop('path') === '/');
  expect(route).toHaveLength(1);
  expect(route.name()).toBe('Route');
});

it('has a unrestricted route for portfolio page', () => {
  const route = routes.filterWhere(n => n.prop('path') === '/portfolio');
  expect(route).toHaveLength(1);
  expect(route.name()).toBe('Route');
});

it('has a public route for login page, with fallback url specified', () => {
  const route = routes.filterWhere(n => n.prop('path') === '/login');
  expect(route).toHaveLength(1);
  expect(route.name()).toBe('Connect(PublicRoute)');
  expect(route.prop('fallbackTo')).toBe('/admin');
});

it('has a private route for admin page, with fallback url specified', () => {
  const route = routes.filterWhere(n => n.prop('path') === '/admin');
  expect(route).toHaveLength(1);
  expect(route.name()).toBe('Connect(PrivateRoute)');
  expect(route.prop('fallbackTo')).toBe('/login');
});

it('has a private route for add image page, with fallback url specified', () => {
  const route = routes.filterWhere(n => n.prop('path') === '/add');
  expect(route).toHaveLength(1);
  expect(route.name()).toBe('Connect(PrivateRoute)');
  expect(route.prop('fallbackTo')).toBe('/login');
});
