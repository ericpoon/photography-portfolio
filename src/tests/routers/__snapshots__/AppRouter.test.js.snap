// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AppRouter 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        header={[Function]}
        path="/portfolio"
      />
      <Connect(PublicRoute)
        component={[Function]}
        exact={true}
        fallbackTo="/admin"
        path="/login"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        exact={true}
        fallbackTo="/login"
        header={[Function]}
        path="/admin"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        exact={true}
        fallbackTo="/login"
        header={[Function]}
        path="/add"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</Router>
`;
